### PR TITLE
Add documentation for deploying node with DevStack and Ironic

### PR DESCRIPTION
**Pull request by skanthed**
### Summary
This pull request adds detailed documentation for deploying a node using Ironic and setting up DevStack with Ironic integration, including steps for creating fake nodes.

### CODE REVIEW

**Comment by skanthed** [comment]
@tzumainn Let me know what else needs to be added or if something needs to be removed.

File: docs/source/developer/index.rst [hunk 1]
```diff
@@ -9,6 +9,11 @@ Developers
       * `python-esileapclient`_
    * `python-esiclient`_
 
+.. toctree::
+   :maxdepth: 2
+
+   node-with-devstack
```
Comment (tzumainn) [inline docs/source/developer/index.rst:15]: Can you rename this to "esi-with-devstack"?

File: docs/source/developer/node-with-devstack.rst [hunk 2]
```diff
@@ -0,0 +1,274 @@
+Deploying Node with DevStack and Ironic
+=======================================
+
+This covers the steps to deploy a node using Ironic and set up DevStack with Ironic integration, including creating fake nodes in Ironic.
+
+.. contents:: Table of Contents
+   :depth: 2
+   :local:
+
+Image That We Used
```
Comment (tzumainn) [inline docs/source/developer/node-with-devstack.rst:10]: I might change this to a "Prerequisites" section, similar to what we have at https://esi.readthedocs.io/en/latest/usage/openshift.html#prerequisites.

File: docs/source/developer/node-with-devstack.rst [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+------------------
+
+For this, we used the Ubuntu 22.04 image to provision a node. You can use any image that is compatible with DevStack. You can download the image from the official Ubuntu website:
+
+[Download Ubuntu 22.04](https://releases.ubuntu.com/22.04/ubuntu-22.04-live-server-amd64.iso)
+
+Deploy the Node with an Image and Attach an External Floating IP
```
Comment (tzumainn) [inline docs/source/developer/node-with-devstack.rst:17]: This is my fault, but we probably don't need this section; most people testing ESI won't have an ESI node available; instead they really just need an Ubuntu bare metal node or VM. I'd change the prerequisites to simply say "VM or bare metal node provisioned with..."

File: docs/source/developer/node-with-devstack.rst [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+This covers the steps to deploy a node using Ironic and set up DevStack with Ironic integration, including creating fake nodes in Ironic.
+
+.. contents:: Table of Contents
+   :depth: 2
+   :local:
+
+Image That We Used
+------------------
+
+For this, we used the Ubuntu 22.04 image to provision a node. You can use any image that is compatible with DevStack. You can download the image from the official Ubuntu website:
```
Comment (tzumainn) [inline docs/source/developer/node-with-devstack.rst:13]: Can you link to a devstack page that lists compatible images? It probably makes sense to also say that the instructions below are ubuntu-specific.

File: docs/source/developer/node-with-devstack.rst [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+[Download Ubuntu 22.04](https://releases.ubuntu.com/22.04/ubuntu-22.04-live-server-amd64.iso)
```
Comment (tzumainn) [inline docs/source/developer/node-with-devstack.rst:15]: rst files don't link this way; you can look at `usage/cli.rst` for examples.

File: docs/source/developer/node-with-devstack.rst [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+
+.. prompt:: bash $
+
+   sudo systemctl restart network-manager
+
+**Update the system:**
+
+.. prompt:: bash $
+
+   sudo yum update -y
```
Comment (tzumainn) [inline docs/source/developer/node-with-devstack.rst:110]: Does ubuntu use `yum`?

File: docs/source/developer/node-with-devstack.rst [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+   sudo yum upgrade -y
+
+**Install Git:**
+
+.. prompt:: bash $
+
+   sudo yum install git -y
+
+Install DevStack and configure it to work with Ironic by following these steps:
```
Comment (tzumainn) [inline docs/source/developer/node-with-devstack.rst:119]: as a title of a section, I'd reduce this to "Install DevStack and Configure Ironic"

**Review by tzumainn — CHANGES_REQUESTED** [review]
This is a good start! I've left some comments; can you also add a section explaining how to install esi-leap?

File: docs/source/developer/esi-with-devstack.rst [hunk 8]
```diff
@@ -0,0 +1,242 @@
+Deploying Node with DevStack and Ironic
```
Comment (tzumainn) [inline docs/source/developer/esi-with-devstack.rst:1]: Maybe "Deploying ESI on DevStack"?

File: docs/source/developer/esi-with-devstack.rst [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+=======================================
+
+This covers the steps to deploy a node using Ironic and set up DevStack with Ironic integration, including creating fake nodes in Ironic.
```
Comment (tzumainn) [inline docs/source/developer/esi-with-devstack.rst:4]: This no longer covers deploying a node, so the scope of the document is creating a development deployment of ESI on DevStack.

File: docs/source/developer/esi-with-devstack.rst [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+   ADMIN_PASSWORD=secret
+   DATABASE_PASSWORD=$ADMIN_PASSWORD
+   RABBIT_PASSWORD=$ADMIN_PASSWORD
+   SERVICE_PASSWORD=$ADMIN_PASSWORD
+
+   # Enable Ironic
+   enable_service ironic-api
+   enable_service ironic-cond
+
+   # More settings needs to be added as needed. For a full example, see the Ironic DevStack Guide.
```
Comment (tzumainn) [inline docs/source/developer/esi-with-devstack.rst:136]: Are these settings required, or optional ("as needed")?

**Review by tzumainn — CHANGES_REQUESTED** [review]
Pretty good! Just a few points of clarification.

File: docs/source/developer/esi-with-devstack.rst (hunk 10 above)
Comment (skanthed) [reply docs/source/developer/esi-with-devstack.rst:136]: Yes they need to add local.conf file from the link mentioned below. I was just adding a snippet here.

Comment (tzumainn) [reply docs/source/developer/esi-with-devstack.rst:136]: Okay - I'd get rid of the text "as needed" then.

**Review by tzumainn — APPROVED** [review]
Looks good - thanks!

**Pull request merged**